### PR TITLE
Debezium tutorial scenario

### PR DESCRIPTION
**Pull request by jpechane**
PR for https://github.com/openshift-evangelists/intro-katacoda/issues/98

### CODE REVIEW

**Comment by GrahamDumpleton** [comment]
You need to have people login in from the command line as a specific user and only then have them create the project. If you don't do that, then they can't login to view the application from the web console. I can't see that you have them logging in explicitly.

Please also provide the URL for this scenario under your katacoda account so it can be review properly. Thanks.

**Comment by jpechane** [comment]
https://www.katacoda.com/jpechane/courses/intro-openshift/debezium-getting-started

**Comment by jpechane** [comment]
Please do not review now - it seems there are issues with `courseData` script

**Comment by jpechane** [comment]
A `debezium` user was added, so the project is now visible in console. I move most of init commands to forground script as it seems it fits better to my usescase.

**Comment by thesteve0** [comment]
Hey Jiri:
I am going to shepherd yours through. Is there a reason why you pop the
console out of katacoda?
I do have some wording suggestions. Do you want me to send those as
comments or fix them in the text?

**Comment by jpechane** [comment]
Hi Steve, I am delighted to have you as the reviewer. Please feel free to fix wordings as you wish. I, as a non-native speaker, would just subtract the value from them.
Could you please explain a little bit more on what do you mean by popping out the console?
Thanks!

**Comment by thesteve0** [comment]
Hey Jiri:
Your project URL is no longer starting for me. It just freezes with
"Preparing your environment, hang tight".

What I did mean is, on the bottom half of the screen, if I click on the
OpenShift Console tab, the console popped open a new tab in Firefox for me.

**Comment by BenHall** [comment]
We currently don't support iframe + editor layout. The reason is there was some confusion between changing files and between the editor/iframe dashboard. 

If someone has a idea of where to put the link to the dashboard I'm happy to implement it.

**Comment by jpechane** [comment]
@thesteve0 I've just tried the URL and met no issues :-(.
Regarding console pop up it is probably as @BenHall said. I just shamelessly copied Swarm lesson and started from it.

**Comment by jpechane** [comment]
I originally planned to to use it more for the future extended scenario - so Iam not sure is is worthy to remove it now and get it back later. But I am definitely open to recommendations for the best user experience.

File: intro-openshift/debezium-getting-started/00-intro.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+In this scenario you will learn more about [Debezium](http://debezium.io/), a project that provides change data capture for MySQL, PostgreSQL and MongoDB databases.
+
+You will deploy a complete end-to-end solution that will capture events from database transaction log and deliver it to a downstream processing via [Apache Kafka](https://kafka.apache.org/) broker.
+
+## What is Debezium? 
+
+![Logo](../../assets/intro-openshift/debezium-getting-started/debezium-logo.png)
+
+[Debezium](http://debezium.io/) is a set of distributed services capture row-level changes in your databases so that your applications can see and respond to those changes.
+Debezium records in a transaction log all row-level changes committed to each database table.
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/00-intro.md]: Debezium records all row-level changes committed to a particular database table in a dedicated message topic.

File: intro-openshift/debezium-getting-started/00-intro.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Each application simply reads the transaction logs their interested in, and they see all of the events in the same order in which they occurred.
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/00-intro.md]: reads the topic(s) they are interested in...

File: intro-openshift/debezium-getting-started/00-intro.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Technically Debezium utilizes [Kafka](https://kafka.apache.org/) streaming platform to distribute events captured from database.
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/00-intro.md]: s/Kafka/Apache Kafka/

File: intro-openshift/debezium-getting-started/00-intro.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+It is a set o plug-ins for [Kafka Connect](https://kafka.apache.org/documentation/#connect) that publishes messages to a [Kafka broker](https://kafka.apache.org/documentation/#uses_messaging).
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/00-intro.md]: s/publishes/publish/

File: intro-openshift/debezium-getting-started/00-intro.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+The minimum components required for skeleton deployment are
+* Kafka broker - consiting of a single [Apache ZooKeeper](https://zookeeper.apache.org/) instance for cluster management and a single node of Kafka broker
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/00-intro.md]: s/consiting/consisting/

File: intro-openshift/debezium-getting-started/01-deploying-a-broker.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+``oc get svc -l app=barnabas``{{execute}}
+
+    NAME                 CLUSTER-IP       EXTERNAL-IP   PORT(S)                      AGE
+    kafka                172.30.180.219   <none>        9092/TCP                     5m
+    kafka-headless       None             <none>        9092/TCP                     5m
+    zookeeper            172.30.220.246   <none>        2181/TCP                     5m
+    zookeeper-headless   None             <none>        2181/TCP,2888/TCP,3888/TCP   5m
+
+**3. Verify the the broker is up and running.**
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/01-deploying-a-broker.md]: double "the"

File: intro-openshift/debezium-getting-started/02-deploying-a-kafka-connect.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        {"class":"org.apache.kafka.connect.file.FileStreamSourceConnector","type":"source","version":"1.0.0"}
+    ]
+
+The Connect node has a `MySqlConnector` connector plugin available.
+
+## Congratulations
+
+You have now successfully executed the second step in this scenario. 
+
+You have successfully deployed Kafka node and configured it to contain Debezium.
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/02-deploying-a-kafka-connect.md]: Technically, it's Kafka Connect that contains the Debezium connectors.

File: intro-openshift/debezium-getting-started/03-streaming-data-from-database.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
...ory.customers.Value","field":"after"},{"type":"struct","fields":[{"type":"string","optional":false,"field":"name"},{"type":"int64","optional":false,"field":"server_id"},{"type":"int64","optional":false,"field":"ts_sec"},{"type":"string","optional":true,"field":"gtid"},{"type":"string","optional":false,"field":"file"},{"type":"int64","optional":false,"field":"pos"},{"type":"int32","optional":false,"field":"row"},{"type":"boolean","optional":true,"field":"snapshot"},{"type":"int64","optional":true,"field":"thread"},{"type":"string","optional":true,"field":"db"},{"type":"string","optional":true,"field":"table"}],"optional":false,"name":"io.debezium.connector.mysql.Source","field":"source"},{"type":"string","optional":false,"field":"op"},{"type":"int64","optional":true,"field":"ts_ms"}],"optional":false,"name":"dbserver1.inventory.customers.Envelope","version":1},"payload":{"before":null,"after":{"id":1004,"first_name":"Anne","last_name":"Kretchmar","email":"annek@noanswer.org"},"source":{"name":"dbserver1","server_id":0,"ts_sec":0,"gtid":null,"file":"mysql-bin.000003","pos":154,"row":0,"snapshot":true,"thread":null,"db":"inventory","table":"customers"},"op":"c","ts_ms":1513066125526}}
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/03-streaming-data-from-database.md]: How about adding a comment "Output formatted for the sake of readability" and showing a properly indented version here?

File: intro-openshift/debezium-getting-started/03-streaming-data-from-database.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    Processed a total of 4 messages
+    
+If we add a new record to the table
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/03-streaming-data-from-database.md]: I think there should always be a ":" in sentences leading to a snippet like this.

File: intro-openshift/debezium-getting-started/99-outro.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+In this scenario you learned about change data capture concept and about how you can leverage Debezium for that purpose.
```
Comment (gunnarmorling) [inline intro-openshift/debezium-getting-started/99-outro.md]: s/about/about the/

**Comment by thesteve0** [comment]
I would only expose as much complexity as you need for *this* scenario. If
you are trying to introduce them to Debezium then keep this as simple and
"awesome" as powerful but don't go down all the advanced scenarios...yet.

If you can get by in this case with just a cli and an openshift web console
then just use that.

**Comment by jpechane** [comment]
Hi, i've incroporated Gunnar's comments (just not colons) and changed the layout to `terminal-frame`. Let's continue in a new PR that is created from branch, not master - https://github.com/openshift-evangelists/intro-katacoda/pull/102

**Pull request closed without merging**